frontend/src/components/EventCard/EventCard.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from "react";
+import DeleteIcon from "@mui/icons-material/Delete";
+import SquareIcon from "@mui/icons-material/Square";
+import ModeEditIcon from "@mui/icons-material/ModeEdit";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import CloseIcon from "@mui/icons-material/Close";
+import DeletePopUpScreen from "./DeletePopUpScreen";
+import { useNavigate } from "react-router-dom";
+import { compareIst } from "../../utils/compareIst";
+
+
+const EventCard = ({
+  selectedEvent,
+  setSelectedEvent,
+  popupPosition,
+  isEventOpen,
+  setIsEventOpen,
+}) => {
+  const navigate = useNavigate();
+  const istDateTimeDisplay = new Date(selectedEvent.start)
+    .toDateString()
+    .split(" ");
+  const [deletePopUp, setDeletePopUp] = useState(false);
+  const [cutThrough, setCutThrough] = useState(false);
+  const closePopUp = () => {
+    setIsEventOpen(false);
+  };
+
+  const deleteEvent = (e) => {
+    e.preventDefault();
+    setIsEventOpen(false);
+    setDeletePopUp(true);
+  };
+
+  const updateEvent = (e) => {
+    e.preventDefault();
+    setIsEventOpen(false);
+    navigate("/update");
+  };
+
+  useEffect(() => {
+    const { date1IST, date2IST } = compareIst(
+      new Date(selectedEvent.start),
+      new Date()
+    );
+
+    if (date1IST < date2IST) {
+      setCutThrough(true);
+    } else {
+      setCutThrough(false);
+    }
+  }, [selectedEvent]);
+  return (
+    <>
+      {isEventOpen && (
+        <>
+          <div
+            className="transform -translate-x-1/2 -translate-y-1/2 z-50 w-1/5 mt-3 "
+            style={{
```

Comment:
you should avoid inline style method for to use css modules or class from library directly if any